src/mjolnir/graphbuilder.cc

```diff
@@ -1028,6 +1029,10 @@ void BuildTileSet(const std::string& ways_file,
 
         graphtile.nodes().back().set_timezone(tz_index);
 
+        // set the density as needed.
+        if (use_urban_tag && node.urban())
+          graphtile.nodes().back().set_density(kMaxDensity);
```

Comment:
its unfortunate that we have to do this here and have density set in two places but there is no way around it